File: RxMVVMJavaJavaTest2/app/src/main/java/net/techpda/rxmvvmjavajavatest2/di/ApplicationComponent.java
package net.techpda.rxmvvmjavajavatest2.base.di;

import android.app.Application;

import net.techpda.rxmvvmjavajavatest2.viewmodel.base.BaseApplication;

import javax.inject.Singleton;

import dagger.BindsInstance;
import dagger.Component;
import dagger.android.AndroidInjector;
import dagger.android.DaggerActivity;
import dagger.android.support.AndroidSupportInjectionModule;

@Singleton
@Component(modules = {ContextModule.class, ApplicationModule.class, AndroidSupportInjectionModule.class, ActivityBindingModule.class})
public interface ApplicationComponent extends AndroidInjector<DaggerActivity> {

    void inject(BaseApplication application);

    @Component.Builder
    interface Builder {

        @BindsInstance
        Builder application(Application application);
        ApplicationComponent build();

    }


}

File: ditest2/app/src/main/java/net/techpda/ditest2/ApplicationComponent.kt
package net.techpda.ditest2

import android.app.Application
import dagger.BindsInstance
import dagger.Component
import dagger.android.support.AndroidSupportInjectionModule
import javax.inject.Singleton

@Singleton
@Component(modules = arrayOf(ContextModule::class, AndroidSupportInjectionModule::class, ActivityBindingModule::class))
interface ApplicationComponent {
    fun inject(application: BaseApplication)
}

@Component.Builder
interface Builder {
    @BindsInstance
    fun application(application: Application): Builder

    fun build(): ApplicationComponent
}